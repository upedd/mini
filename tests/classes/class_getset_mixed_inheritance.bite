native print;

class Foo {
    x = 10;

    print_x() {
        print(x);
    }
}

class Bar : Foo {
    _x;
    override get x {
        print("bar.x");
        return _x;
    }

    override set x(value) {
        print("=bar.x");
        print(value);
        _x = value;
    }
}

let instance = Bar();
instance.print_x();
instance.x = 20;
instance.print_x();