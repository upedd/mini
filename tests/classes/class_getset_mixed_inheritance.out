=bar.x
10
bar.x
10
=bar.x
20
bar.x
10
